README.md

```diff
@@ -1,133 +1,416 @@
+
+### Source installation
+
+Prerequisites:
+* `libsqlite3-dev`, especially `sqlite.h`
+* `gcc`
+* `make`
+
+#### 1. Install SQLite library
+
+For Debian or Ubuntu:
+`apt-get install libsqlite3-dev`
+
+You can also [download SQLite source code][1] and [build SQLite][2].
+
+#### 2. Build and install sqlite_fdw
+
+Add a directory of `pg_config` to PATH and build and install `sqlite_fdw`.
+
+```sh
 make USE_PGXS=1
 make install USE_PGXS=1
-</pre>
+```
 
-If you want to build sqlite_fdw in a source tree of PostgreSQL, use
-<pre>
+If you want to build `sqlite_fdw` in a source tree of PostgreSQL, use
+```sh
 make
 make install
-</pre>
+```
+
+Usage
+-----
+
+## CREATE SERVER options
+
+`sqlite_fdw` accepts the following options via the `CREATE SERVER` command:
+
+- **database** as *string*, **required**
+
+  SQLite database path.
+
+- **truncatable** as *boolean*, optional
+
+  Allows foreign tables to be truncated using the `TRUNCATE` command.
+  
+- **keep_connections** as *boolean*, optional
+  
+  Allows to keep connections to SQLite while there is no SQL operations between PostgreSQL and SQLite.
+  
+- **batch_size** as *integer*, optional
+
+  Specifies the number of rows which should be inserted in a single `INSERT` operation. This setting can be overridden for individual tables.
+
+
+## CREATE USER MAPPING options
+
+`sqlite_fdw` no need any `CREATE USER MAPPING` command.
+
+Just ensure `postgres` OS user have permissions for reading or writing on SQLite database file.
+
+
+## CREATE FOREIGN TABLE options
+
+`sqlite_fdw` accepts the following table-level options via the
+`CREATE FOREIGN TABLE` command:
+
+- **table** as *string*, **required**
+
+  SQLite table name.
+
+- **truncatable** as *boolean*, optional
+  
+  Allows table to be truncated using the `TRUNCATE` command.
+
+- **batch_size** as *integer*, optional
+
+  See `CREATE SERVER` options section for details.  
+  
+`sqlite_fdw` accepts the following column-level options via the
+`CREATE FOREIGN TABLE` command:
+
+- **column_name** as *string*, optional
+
+  This option gives the column name to use for the column on the remote server.
+
+- **column_type** as *string*, optional
+
+  Option to convert INT SQLite column (epoch Unix Time) to be treated/visualized as TIMESTAMP in PostgreSQL.
+
+- **key** as *boolean*, optional
+
+  Indicates a column as a part of primary key or unique key of SQLite table.
+  
+## IMPORT FOREIGN SCHEMA options
+
+`sqlite_fdw` supports [IMPORT FOREIGN SCHEMA](https://www.postgresql.org/docs/current/sql-importforeignschema.html)
+(PostgreSQL 9.5+) and accepts no custom options for this command.
+
+## TRUNCATE support
+
+`sqlite_fdw` implements the foreign data wrapper `TRUNCATE` API, available
+from PostgreSQL 14.
+
+As SQlite does not provide a `TRUNCATE` command, it is simulated with a
+simple unqualified `DELETE` operation.
+
+Following restrictions apply:
 
-## Usage
+ - `TRUNCATE ... CASCADE` is not supported
```

Comment:
Actually, TRUNCATE ... CASCADE can be simulated if we create child table of SQLite with foreign keys and ON DELETE CASCADE, and then executing TRUNCATE (which will be deparsed to DELETE).
It has the same affect as TRUNCATE ... CASCADE.
Therefore, I think describing it as "not supported" is not correct.

Reply:
Thanks, @son-phamngoc. Corrected.